src/item/abstract_item.cpp

```diff
@@ -450,14 +450,28 @@ QPen AbstractItem::connectorStyle(int type)
     return Resource::get(QString("item_connector_pen%1").arg(type), QPen(Qt::black, connectorHeight()));
 }
 
-void AbstractItem::remove()
+void AbstractItem::disconnectConnections() {
+    Q_D(AbstractItem);
+    for (int i = d->_inputs.count() - 1; i >= 0; i--) {
```

Comment:
I'd suggest using `std::for_each` with `QList::rbegin` to prevent errors related to the index variable `i` and also to make the code more easily understandable.

Maybe also add a comment explaining why this needs to be done in reverse order.